test/lit/passes/optimize-instructions-call_ref.wast

```diff
@@ -253,29 +256,42 @@
  ;; CHECK:      (func $call_ref-to-select (type $5) (param $x i32) (param $y i32) (param $z i32) (param $f (ref $i32_i32_=>_none))
  ;; CHECK-NEXT:  (local $4 i32)
  ;; CHECK-NEXT:  (local $5 i32)
+ ;; CHECK-NEXT:  ;;@ file.cpp:20:2
```

Comment:
How did these `block`s and `local.get`s get the debug info? The code only seems to put the debug function for the new calls...

Reply:
The outer block gets it because we replace the original instruction with the block, so the debuginfo is copied by default in `replaceCurrent` (which seems like a reasonable default, and I don't think it is harmful here). But I am actually not sure why the `local.get` also receives it... we must have specific code for that somewhere. It also seems ok to me though.

Reply:
But the debug info was already being copied in `replaceCurrent`, no? I wonder what caused the change.

Reply:
Ok, I think I figured it out: these `local.get`s are not actually new. They are reused from the original code before this optimization, and they all had debuginfo before, so they still have it after. CallUtils adds some additional `local.get`s, which was confusing, and those have no debug info as expected.

Reply:
Specifically `$x,$y,$z` are pre-existing locals and gets, with names and debug info, and `$4,$5` are new locals with neither names nor debug info.

Reply:
> But the debug info was already being copied in `replaceCurrent`, no? I wonder what caused the change.

But why did the `block` get new debug info then? `replaceCurrent` was already coping the debug info. Also I still don’t get why `local.get`s that already existed newly got debug info. If those `local.get`s already had debug info why was it not shown before? Other than `makeCall`, this PR does not change other things semantically, no?

Reply:
The block gets debug info after this PR because the test had no debug info before. See line 310. Maybe a separate NFC PR that just modified the test would have made that clearer, sorry.

The `local.get`s had debug info before, but by default we don't repeat debug info in the printed output:
```wat
  ;;@ file.cpp:20:2
  (call_ref $i32_i32_=>_none
   (local.get $x)
   (local.get $y)
   (select
    (ref.func $foo)
    (ref.func $bar)
    (local.get $z)
   )
  )
```
Not only the `call_ref` has that debug location but all its children. We just don't print it (unless `BINARYEN_PRINT_FULL=1` is set), because typically nested children have the same locations, so this makes the default text easier to read.

(If a child had different debug info we'd print that, or if it was marked as having no debug info we'd add `;;@`, with nothing else on that line.)

Reply:
Ah I see. Also I missed the newly added debug info. Thanks for the explanation!